internal/controller/logger_manager.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2024 Illumio, Inc. All Rights Reserved.
+
+package controller
+
+import (
+	"io"
+	"os"
+	"sync"
+	"time"
+
+	pb "github.com/illumio/cloud-operator/api/illumio/cloud/k8scluster/v1"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"google.golang.org/grpc/connectivity"
+)
+
+const (
+	maxBufferSize = 100
+	flushInterval = 5 * time.Second
+)
+
+type ClientConnInterface interface {
+	GetState() connectivity.State
+	Close() error
+}
+
+// BufferedGrpcWriteSyncer is a custom zap writesync that writes to a grpc stream
+// In case stream is not connected it will buffer to memory
+type BufferedGrpcWriteSyncer struct {
+	client   pb.KubernetesInfoService_SendLogsClient
+	conn     ClientConnInterface
+	buffer   []*zapcore.Entry
+	mutex    sync.Mutex
+	done     chan struct{}
+	logger   *zap.SugaredLogger
+	logLevel zap.AtomicLevel
+	encoder  zapcore.Encoder
+}
+
+// NewBufferedGrpcWriteSyncer returns a new BufferedGrpcWriteSyncer
+func NewBufferedGrpcWriteSyncer(client pb.KubernetesInfoService_SendLogsClient, conn ClientConnInterface) *BufferedGrpcWriteSyncer {
```

Comment:
Since this is always passed `(nil, nil)`, you could remove those parameters altogether.
```suggestion
func NewBufferedGrpcWriteSyncer() *BufferedGrpcWriteSyncer {
```